### PR TITLE
[ACR] Add swagger for 2017-10-01 api-version

### PR DESCRIPTION
**Pull request by djyou**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/2017-10-01/containerregistry.json [hunk 1]
```diff
@@ -0,0 +1,2043 @@
...
+    "Sku": {
+      "description": "The SKU of a container registry.",
+      "required": [
+        "name"
+      ],
+      "properties": {
+        "name": {
+          "description": "The SKU name of the container registry. Required for registry creation.",
+          "enum": [
+            "Classic",
```
Comment (djyou) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/2017-10-01/containerregistry.json]: Changed from `Basic`.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/2017-10-01/containerregistry.json [hunk 2]
```diff
@@ -0,0 +1,2043 @@
...
+          "type": "string",
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": true
+          }
+        },
+        "tier": {
+          "description": "The SKU tier based on the SKU name.",
+          "enum": [
+            "Classic",
```
Comment (djyou) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/2017-10-01/containerregistry.json]: Changed from `Basic`.

**Comment by djyou** [comment]
Besides the above commented change in SKU name, this new api-version is the same as [2017-06-01-preview](https://github.com/Azure/azure-rest-api-specs/blob/current/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/2017-06-01-preview/containerregistry.json) version. /cc @SajayAntony

**Comment by SajayAntony** [comment]
@amarzavery - to avoid breaking existing customers should we change the CLI to accept both Basic/Classic?

**Comment by amarzavery** [comment]
Yes this would affect customers. Does the API accept old values?
/cc @johanste for CLI breaking changes
@ravbhatnagar  - for reviewing API breaking changes.

**Comment by fearthecowboy** [comment]
> @SajayAntony :
>> to avoid breaking existing customers should we change the CLI to accept both Basic/Classic?

>@amarzavery :
>> Yes this would affect customers. Does the API accept old values?

And if it doesn't accept the old value, could that be done?

**Comment by djyou** [comment]
We planned to support both `Classic` and `Basic` Az CLI and Powershell cmdlets to avoid breaking existing users.
If this new api-version accepts `Basic`, do we put it in swagger or just secretly accept it so existing users won't break?

**Comment by johanste** [comment]
Renaming properties does cause customer pain and introduce friction for developers to move to new API versions. 

For this specific case, while it would be possible to add Basic as a valid value, it would only solve the request/input side of the problem. Any client that relied/tried to make sense of the old value (Basic) in a response would still miss the fact that the value would presumably now be 'Classic' instead. This is true for the proposed overload of the PowerShell cmdlets as well - I don't see how it would address any now broken code that relied on the return value being 'Basic' for the SKU.

So, my recommendation is that, **if we have decided that the change is important enough that it outweighs the fact that clients will have to possibly update their code in order to make use of the new API version**, we don't put the old value (Basic) in the swagger for the new API version.

**Comment by selvasingh** [comment]
@milismsft will also review this PR

**Comment by djyou** [comment]
@ravbhatnagar We have been having discussions on this change for a while and the team has decided that the change to have `Classic` in the new api-version is important enough. It would be a breaking change at any point of time so sooner seems better than later.

As @johanste also commented, we would prefer to have `Classic` only but `Basic` can be accepted too if ARM requires. Thanks. /cc @SajayAntony

**Comment by johanste** [comment]
@djyou, you would probably want to ensure that the service continues to return 'Basic' for the name for calls that use the old API version even though you switch to 'Classic' for calls using this API version. Or else you risk breaking existing clients using the older API version...

**Comment by djyou** [comment]
@johanste Correct, `Classic` will only be accepted/returned for this new api-version. Even if a resource is created in `2017-10-01` using `Classic`, a GET using previous api-versions will still return `Basic`.

**Comment by ravbhatnagar** [comment]
The change looks good to me. I agree if the scenario is important, then just move to the newly supported value and dont include basic in the new api-version. ARM Signs off.

**Review by fearthecowboy — APPROVED** [review]

**Comment by fearthecowboy** [comment]
Done.

**Pull request merged**